starknet_py/serialization/data_serializers/enum_serializer.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import Dict, Generator
+
+from indexed import IndexedOrderedDict
+
+from starknet_py.serialization._context import (
+    DeserializationContext,
+    SerializationContext,
+)
+from starknet_py.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.serialization.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class EnumSerializer(CairoDataSerializer[Dict, TupleDataclass]):
+    """
+    Serializer of enums.
+    Can serialize a dictionary.
```

Comment:
What about a named tuple and a tuple dataclass? Could the serialization of them be implemented?

Reply:
Do we really want it? Structures are serialized from dicts only